src/proxy/momento/src/listener.rs

```diff
@@ -17,7 +18,7 @@ pub(crate) async fn listener(
         if let Ok((socket, _)) = listener.accept().await {
             TCP_ACCEPT.increment();
 
-            let client = client_builder.clone().build();
+            let client = client.clone();
```

Comment:
Does cloning the client reuse the grpc channel or create a new one? If the former, we may run into the concurrent session limit when more than 100 connections to the proxy use the same cache.

Reply:
> Does cloning the client reuse the grpc channel or create a new one? If the former, we may run into the concurrent session limit when more than 100 connections to the proxy use the same cache.

It does re-use the same channel.

> I think the earlier approach of creating a dedicated momento client for each connection to the proxy is better.

I will make this change.

Reply:
@krispraws I made this change in [9c07408](https://github.com/pelikan-io/pelikan/pull/142/commits/9c07408bc1524424fc187e3df1b2fd35d6c11e0c).